shared/services/wallet/masquerade-wallet.go

```diff
@@ -0,0 +1,246 @@
+	pm         *passwords.PasswordManager
+	am         *AddressManager
+	encryptor  *eth2ks.Encryptor
+	chainID    *big.Int
+
+	// Encrypted store
+	ws *walletStore
+
+	// Seed & master key
+	seed []byte
+	mk   *hdkeychain.ExtendedKey
+
+	// Node key cache
+	nodeKey     *ecdsa.PrivateKey
+	nodeKeyPath string
+
+	// Desired gas price & limit from config
+	maxFee         *big.Int
+	maxPriorityFee *big.Int
+	gasLimit       uint64
+}
+
+// Getter to check if node is masquerading
+func (w *masqueradeWallet) IsNodeMasquerading() bool {
+	return true
+}
+
+// Gets the derived wallet address, if one is loaded. Returns empty address if wallet isn't loaded.
+func (w *masqueradeWallet) GetAddress() (common.Address, error) {
+
+	// Return if wallet is uninitialized
+	if !(w.ws != nil && w.seed != nil && w.mk != nil) {
+		return common.Address{}, nil
+	}
+
+	// Get private key
+	privateKey, _, err := w.getNodePrivateKey()
+	if err != nil {
+		return common.Address{}, nil
+	}
+
+	// Get public key
+	publicKey := privateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	if !ok {
+		return common.Address{}, nil
+	}
+	return crypto.PubkeyToAddress(*publicKeyECDSA), nil
+}
+
+// Change the node's effective address to a different one. Node and watchtower tasks will continue to run normally using the loaded wallet.
+func (w *masqueradeWallet) MasqueradeAsAddress(newAddress common.Address) error {
+	return w.am.SetAndSaveAddress(newAddress)
+}
+
+// End a masquerade, restoring your node's effective address back to your wallet address if one is loaded
+func (w *masqueradeWallet) EndMasquerade() error {
+	if w.am == nil {
+		return errors.New("node is not masquerading")
+	}
+
+	return w.am.DeleteAddressFile()
+}
+
+// Gets the wallet's chain ID
+func (w *masqueradeWallet) GetChainID() *big.Int {
+	copy := big.NewInt(0).Set(w.chainID)
+	return copy
+}
+
+// Add a keystore to the wallet
+func (w *masqueradeWallet) AddKeystore(name string, ks keystore.Keystore) {
+	return
+}
+
+// Always return true as we're masquerading
+func (w *masqueradeWallet) IsInitialized() bool {
+	return true
+}
+
+// Attempt to initialize the wallet if not initialized and return status
+func (w *masqueradeWallet) GetInitialized() (bool, error) {
+	if w.IsInitialized() {
+		return true, nil
+	}
+	return w.loadStore()
+}
+
+// Serialize the wallet to a JSON string
+func (w *masqueradeWallet) String() (string, error) {
+	return "", ErrIsMasquerading
+
+}
+
+// Initialize the wallet from a random seed
+func (w *masqueradeWallet) Initialize(derivationPath string, walletIndex uint) (string, error) {
+	return "", ErrIsMasquerading
+
+}
+
+// Recover a wallet from a mnemonic
+func (w *masqueradeWallet) Recover(derivationPath string, walletIndex uint, mnemonic string) error {
+	return ErrIsMasquerading
+
+}
+
+// Recover a wallet from a mnemonic - only used for testing mnemonics
+func (w *masqueradeWallet) TestRecovery(derivationPath string, walletIndex uint, mnemonic string) error {
```

Comment:
i think ErrIsMasquerading is also appropriate here. We don't let you call Recover while masquerading, so TestRecovery should also fail